core/src/main/java/org/mapfish/print/map/image/ImageLayer.java

```diff
@@ -0,0 +1,244 @@
+      * <p>Type: <code>image</code></p>
+      */
+    public static final class ImageLayerPlugin extends AbstractGridCoverageLayerPlugin implements MapLayerFactoryPlugin<ImageParam> {
+        private static final String TYPE = "image";
+        @Autowired
+        private ForkJoinPool forkJoinPool;
+        @Autowired
+        private StyleParser styleParser;
+        @Autowired
+        private MetricRegistry metricRegistry;
+
+        @Override
+        public Set<String> getTypeNames() {
+            return Collections.singleton(TYPE);
+        }
+
+        @Override
+        public ImageParam createParameter() {
+            return new ImageParam();
+        }
+
+        @Nonnull
+        @Override
+        public ImageLayer parse(@Nonnull final Template template,
+                              @Nonnull final ImageParam layerData) throws Throwable {
+
+            String styleRef = layerData.style;
+            return new ImageLayer(this.forkJoinPool,
+                    super.<GridCoverage2D>createStyleSupplier(template, styleRef),
+                    layerData, this.metricRegistry);
+        }
+    }
+
+    /**
+     * The parameters for reading an image file, either from the server or from a URL.
+     */
+    public static final class ImageParam extends AbstractLayerParams {
+         
+         /**
+         * The base URL for the image file.  Used for making request.
+         */
+         public String baseURL;
+         
+         /**
+          * The extent of the image.  Used for placing image on map.
+          */
+         public double[] extent;
+    
+         /**
+          * The styles to apply to the image.
+          */
+        @HasDefaultValue
+        public String style = Constants.Style.Raster.NAME;
+
+
+        public String getBaseUrl() {
+            return this.baseURL;
+        }
+    }
+
+    @Override
+    protected BufferedImage loadImage(final MfClientHttpRequestFactory requestFactory, 
+              final MapfishMapContext transformer) throws Throwable {
+        final ImageParam layerParam = this.params;
+        final URI commonUri = new URI(layerParam.getBaseUrl());
+        final int maxYIndex = 3;
+        final Double extentMinX = layerParam.extent[0];
+        final Double extentMinY = layerParam.extent[1];
+        final Double extentMaxX = layerParam.extent[2];
+        final Double extentMaxY = layerParam.extent[maxYIndex];
+        
+        final Rectangle paintArea = transformer.getPaintArea();
+        
+        final ReferencedEnvelope envelope = transformer.getBounds().toReferencedEnvelope(paintArea, transformer.getDPI());
+        final CoordinateReferenceSystem mapProjection = envelope.getCoordinateReferenceSystem();
+        
+        
+        //File imageFile = File.createTempFile("downloadedimage", "png");
+        Closer closer = Closer.create();
+        final BufferedImage bufferedImage = new BufferedImage(paintArea.width, paintArea.height, TYPE_INT_ARGB_PRE);
```

Comment:
Looking at this code, I am wondering if it would not be better to inherit from [AbstractSingleImageLayer](https://github.com/mapfish/mapfish-print/blob/master/core/src/main/java/org/mapfish/print/map/image/AbstractSingleImageLayer.java), similar to [WmsLayer](https://github.com/mapfish/mapfish-print/blob/master/core/src/main/java/org/mapfish/print/map/image/wms/WmsLayer.java). What do you think?